database/database.go

```diff
@@ -55,22 +56,24 @@ func (db *Database) refreshDbConfig(ahr *httpUtils.AuthenticatedHttpRequester) {
 }
 
 func (db *Database) Replicate(shard, replica string, ahr *httpUtils.AuthenticatedHttpRequester) error {
-	replicaNode := fmt.Sprintf("couchdb@%s", replica)
+	replicaNode := node.At(replica)
 
-	if sliceUtils.Contains(db.config.ByNode[replicaNode], shard) {
-		return fmt.Errorf("%s is already replicating %s", replicaNode, shard)
+	if sliceUtils.Contains(db.config.ByNode[replicaNode.GetAddr()], shard) {
```

Comment:
`replicaNode.GetAddr()` use of `Get` is not idiomatic in Go. Use `.Addr()` instead.